lib/guardian.ex

```diff
@@ -343,7 +343,7 @@ defmodule Guardian do
       # Provide a way to get at the configuration during compile time
       # for other macros that may want to use them
       @config fn ->
-        the_otp_app |> Application.get_env(__MODULE__, []) |> Keyword.merge(the_opts)
+        the_otp_app |> Application.compile_env(__MODULE__, []) |> Keyword.merge(the_opts)
```

Comment:
This is weird thou because this is not the module scope. The function is inside an anonymous closure.

Is this a false promise?

Reply:
Hey @yordis , I'm not sure I understand what you mean by a `false promise` here?!

We started getting this warning from a module within our application that just has `use Guardian, otp_app: :our_app`.
This was the line getting failing the compilation when running `mix compile --warnings-as-errors`.

The rest of the module is just some wrapper functions around `Guardian.DB.on_refresh` and such.

Let me know if I can provide more info here.

Cheers.

Reply:
I checked with Jose about it, and we are safe to go.

Let me do something else before we merge this one first 🙏🏻

Reply:
Let me know if you need help with anything else here @yordis .

Cheers.

Reply:
I recently added the following https://github.com/ueberauth/.github/blob/master/SECURITY.md

I am upgrading all the packages under Ueberauth to follow the supported versions. A lot of grunt work 😭 

We have a misleading setup at the moment and want to ensure people don't rely on unmaintained Elixir versions and whatnot.

Currently, I have an aggressive deadline, so I am trying my best to little by little do them.

What to do?

1. Update the `mix.exs` to reflect the supported Elixir versions
2. Bump the `mix.exs` to reflect such changes
3. Update the CHANGELOG.md entry
4. Create PR
5. Release the package
6. Repeat for every package

Reply:
@crova do you mind following the https://github.com/ueberauth/.github/blob/master/SECURITY.md and the steps I shared before?